### PR TITLE
Project changes

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Make current private. Outside of project it was being accessed to get name and id. Create getters for both of these.

I then also tried to document what members we can expect on the current. One of these (.level) I managed to remove. One other of these (.hidden) it's not clear to me where/how it is set.

I ran through a couple basic project scenarios. Do we have any test coverage for project? (I think the answer is no). For the most part, these changes don't feel very scary to me.  Most interesting would be if you can think of anywhere that we access project that I wouldn't have found by searching for "project."

### CODE REVIEW

File: shared/js/project.js [hunk 1]
```diff
@@ -14,30 +14,65 @@ var events = {
   hashchange: 'hashchange'
 };
 
+/**
+ * @typedef {Object} ProjectInstance
+ * @property {string} id
+ * @property {string} name
+ * @property {string} levelHtml
+ * @property {string} levelSource
+ * hidden // unclear when this ever gets set
```
Comment (Bjvanminnen) [inline shared/js/project.js]: Where does this get set?

Comment (joshlory) [inline shared/js/project.js]: I think this comes down from the Channels API.

Comment (Bjvanminnen) [inline shared/js/project.js]: I wasn't seeing it get set. Do you know in what scenario it will be?

**Comment by joshlory** [comment]
LGTM!  Probably worth checking a few project share scenarios, such as from an embedded Applab level from inside one of the CSP scripts.

**Comment by Bjvanminnen** [comment]
@joshlory While testing things, I found a couple issues. Could you take another quick look at my latest changes when you have a chance?

**Pull request merged**